yb-voyager/src/queryissue/unsupported_ddls.go

```diff
@@ -0,0 +1,617 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queryissue
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/samber/lo"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/issue"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/queryparser"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/srcdb"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+// DDLIssueDetector interface defines methods for detecting issues in DDL objects
+type DDLIssueDetector interface {
+	DetectIssues(queryparser.DDLObject) ([]issue.IssueInstance, error)
```

Comment:
Do we really need to do 1. Parse into DDLObject, 2. pass DDLObject to Detector? 

Can we just pass query directly to each detector, let each detector parse accordingly however it wants ? 

You wouldn't need a `GetDDLParser` function because each detector is doing it's own parsing. 
You just need a`GetDDLDetector` that can just be based on the nodeType, rather than the DDLObject type.

Reply:
An interface like `DDLObject` is useful when you are operating on that interface type directly (like if a caller calls `DDLObject().GetSchemaName(), but here as far as I can see, all uses of DDLObject first typecast it to the concrete type. So, it doesn't really make sense then to have a DDLObject interface. 

Correct me if my assumption is wrong..

Reply:
Yes currently, not using it like that when I am calling the `DDLObject.GetObjectName()` because wherever I am using the DDLObject I have to typecast it anyway to the specific type for other information of that type, but as I said we can utilize this in other areas where we could utilize this and get the object names directly from the DDLObject for various DDL types. 
One such instance, I added one function for getting objectName from the parser for INDEX, TRIGGER, and POLICY for the schema summary of analyze/assess so instead of that function we could directly get the objectName with this interface.
```
obj := ParseAndProcessDDL(ddl)
obj.GetObjectName()
```